### ISSUE TITLE
FAQ on service fee

### ISSUE THREAD

**Issue by toutantic** [PrestaShop/docs#465]
Question in need for an answer that should go here:
https://devdocs.prestashop.com/1.7/faq/shipping/

## How do I add a service fee to an order? 
I want to create a module that provides an additional service (insurance, tracking, …) to an order. The service is billed to the buyer not the merchant.
How do I add a service fee to an order ?

**Comment by syffs** [comment]
any insight on this subject ?

**Comment by syffs** [comment]
on this topic, one would need to know:
- how to modify the total amount of the cart: I found that [this module](https://addons.prestashop.com/fr/paiement-carte-wallet/19176-paygreen-solution-de-paiement.html) does just that.
- how to edit checkout summary to add a custom line like this:
**2 items** 20€
**shipping** free
<ins>**my service** 5€</ins>
...
TOTAL 25€
- how to edit generated content like email/bill with the same information
- anything else I'm missing ?
any contribution would be appreciated :smiley:

**Comment by syffs** [comment]
@toutantic could you please assist, I'd be happy to submit a PR with at least some pointers ?

**Comment by Quetzacoalt91** [comment]
Hi,

Does your service fee has a fixed price? If yes, you may create a virtual product. Then you may add it in the cart during the checkout process.
This won't appear separately from the other products, but this fee will be taken in account.

**Comment by syffs** [comment]
Hello, Thanks for taking the time !

No, my service would have a dynamic price...

**Comment by matthieu-rolland** [comment]
Hello @syffs , I have some questions about what you're trying to achieve.

How will the service price be applied, and how its amount will be calculated?  

Let's say I'm the customer, I do my order in FO, will I see it on the order summary already? If yes, how was the amount of the service defined? Is it related to specific products, or to the total amount of the order? Is it a percentage?

**Comment by syffs** [comment]
Hi @matthieu-rolland, thanks for picking this up !

Here's an end-user use case:
1. I'm browsing my favorite merchant's prestashop website
2. I end up on the detailed product page that I want to buy
3. From this product page
    a. there happens to be a script of our service which may or may not display a widget offering a service (depending on elligibility) that'll be attached to this product
    b. The pricing of the service is dynamically calculated in our backend based multiple factors
4. I add the product to the cart along with our service
5. from the cart page, I can
    a. see all of my products with (if relevant) their attached service
    b. have detailed pricing of all the cart items (including both products and services)
    c. add/remove a service to an elligible product
6. I complete my order and pay the total amount including both products and services
7. generated summary email/bill include everything as well

I'd be glad to discuss more specific ideas over the phone !

**Comment by matthieu-rolland** [comment]
@syffs Yes, we should talk about this !  we have a community slack for the open source project:

**prestashop.slack.com**

Can you join us over there? My nickname is the same as here: matthieu-rolland
